### PR TITLE
Replace Security Key with Passkey in WebAuthn UIs and their documents

### LINKED ISSUE #27147: Replace "Security Key" with "Passkey" in WebAuthn UIs and their documents
### Description

WebAuthn UIs (authentication, policy) and documentation about WebAuthn use the word "Security Key".
Recently, "Passkey" becomes used commonly instead of "Security Key".
Therefore, the issue is for replacing the word "Security Key" with "Passkey".

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by tnorimat**
closes #27147

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@tnorimat Thanks!

**Comment by mposolda** [comment]
@keycloak/ui-maintainers Anyone available to review UI changes in this PR please?

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java [hunk 1]
```diff
@@ -262,7 +262,7 @@ public void webAuthnPasswordlessAlternativeWithWebAuthnAndPassword() throws IOEx
                     .client("account")
                     .assertEvent();
 
-            // Password + WebAuthn security key
+            // Password + WebAuthn passkey
```
Comment (jonkoops) [inline testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java:265]:
```suggestion
            // Password + WebAuthn Passkey
```

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java [hunk 2]
```diff
@@ -409,19 +409,19 @@ private void testWebAuthn(boolean passwordless) {
 
         if (passwordless) {
             credentialType = webAuthnPwdlessCredentialType;
-            expectedHelpText = "Use your security key for passwordless sign in.";
+            expectedHelpText = "Use your passkey for passwordless sign in.";
```
Comment (jonkoops) [inline testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java:412]:
```suggestion
            expectedHelpText = "Use your Passkey for passwordless sign in.";
```

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java [hunk 3]
```diff
@@ -409,19 +409,19 @@ private void testWebAuthn(boolean passwordless) {
...
             providerId = WebAuthnPasswordlessRegisterFactory.PROVIDER_ID;
         } else {
             credentialType = webAuthnCredentialType;
-            expectedHelpText = "Use your security key to sign in.";
+            expectedHelpText = "Use your passkey to sign in.";
```
Comment (jonkoops) [inline testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java:416]:
```suggestion
            expectedHelpText = "Use your Passkey to sign in.";
```

File: themes/src/main/resources/theme/base/login/messages/messages_en.properties [hunk 4]
```diff
@@ -473,25 +473,25 @@ recovery-codes-download-file-date=These codes were generated on
...
-webauthn-help-text=Use your security key to sign in.
-webauthn-passwordless-display-name=Security Key
-webauthn-passwordless-help-text=Use your security key for passwordless sign in.
-webauthn-login-title=Security Key login
-webauthn-registration-title=Security Key Registration
-webauthn-available-authenticators=Available Security Keys
+webauthn-display-name=Passkey
+webauthn-help-text=Use your passkey to sign in.
+webauthn-passwordless-display-name=Passkey
+webauthn-passwordless-help-text=Use your passkey for passwordless sign in.
```
Comment (jonkoops) [inline themes/src/main/resources/theme/base/login/messages/messages_en.properties:479]:
```suggestion
webauthn-passwordless-help-text=Use your Passkey for passwordless sign in.
```

File: themes/src/main/resources/theme/keycloak.v2/account/messages/messages_en.properties [hunk 5]
```diff
@@ -106,10 +106,10 @@ recovery-authn-codes-help-text=These codes can be used to regain your access in
...
 recovery-codes-number-remaining={0} recovery codes remaining
 recovery-codes-generate-new-codes=Generate new codes to ensure access to your account
-webauthn-display-name=Security key
-webauthn-help-text=Use your security key to sign in.
-webauthn-passwordless-display-name=Security key
-webauthn-passwordless-help-text=Use your security key for passwordless sign in.
+webauthn-display-name=Passkey
+webauthn-help-text=Use your passkey to sign in.
+webauthn-passwordless-display-name=Passkey
+webauthn-passwordless-help-text=Use your passkey for passwordless sign in.
```
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/messages/messages_en.properties:112]:
```suggestion
webauthn-passwordless-help-text=Use your Passkey for passwordless sign in.
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/account-ui/maven-resources/theme/keycloak.v3/account/messages/messages_en.properties [hunk 6]
```diff
@@ -180,8 +180,8 @@ unLinkSuccess=Successfully unlinked account
 applications=Applications
 sharedWithMe=Shared with Me
 username=Username
-webauthn-display-name=Security Key
-webauthn-help-text=Use your security key to sign in.
-webauthn-passwordless-display-name=Security Key
-webauthn-passwordless-help-text=Use your security key for passwordless sign in.
+webauthn-display-name=PassKey
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply js/apps/account-ui/maven-resources/theme/keycloak.v3/account/messages/messages_en.properties:183]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 7]
```diff
@@ -66,18 +66,18 @@ image:images/account-console-signing-in.png[Signing in]
 
 . Click *Signing In*.
 
-. Click *Set up Security Key*.
+. Click *Set up Passkey*.
 +
 .Signing In
-image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Security Key]
+image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Passkey]
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:72]: I fixed it.

[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:72]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 8]
```diff
@@ -66,18 +66,18 @@ image:images/account-console-signing-in.png[Signing in]
 
 . Click *Signing In*.
 
-. Click *Set up Security Key*.
+. Click *Set up Passkey*.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:69]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 9]
```diff
@@ -66,18 +66,18 @@ image:images/account-console-signing-in.png[Signing in]
...
 +
 .Signing In
-image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Security Key]
+image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Passkey]
 
-. Prepare your WebAuthn Security Key. How you prepare this key depends on the type of WebAuthn security key you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
+. Prepare your WebAuthn Passkey. How you prepare this key depends on the type of WebAuthn passkey you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:74]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 10]
```diff
@@ -91,18 +91,18 @@ image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Sec
 
 . Click *Signing In*.
 
-. Click *Set up Security Key* in the *Passwordless* section.
+. Click *Set up Passkey* in the *Passwordless* section.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:94]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 11]
```diff
@@ -91,18 +91,18 @@ image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Sec
...
 +
 .Signing In
-image:images/account-console-signing-in-webauthn-passwordless.png[Signing In With Security Key]
+image:images/account-console-signing-in-webauthn-passwordless.png[Signing In With Passkey]
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:97]: I fixed it. I will update the screenshot.

[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:97]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 12]
```diff
@@ -91,18 +91,18 @@ image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Sec
...
 
-. Prepare your WebAuthn Security Key. How you prepare this key depends on the type of WebAuthn security key you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
+. Prepare your WebAuthn Passkey. How you prepare this key depends on the type of WebAuthn passkey you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc:99]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 13]
```diff
@@ -162,7 +162,7 @@ The appropriate method to register a WebAuthn authenticator depends on whether t
 
 ===== New user
 
-If the *WebAuthn Register* required action is *Default Action* in a realm, new users must set up the WebAuthn security key after their first login.
+If the *WebAuthn Register* required action is *Default Action* in a realm, new users must set up the WebAuthn passkey after their first login.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:165]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 14]
```diff
@@ -191,7 +191,7 @@ After successful registration, the user's browser asks the user to enter the tex
 
 {project_name} uses WebAuthn for two-factor authentication, but you can use WebAuthn as the first-factor authentication. In this case, users with `passwordless` WebAuthn credentials can authenticate to {project_name} without a password. {project_name} can use WebAuthn as both the passwordless and two-factor authentication mechanism in the context of a realm and a single authentication flow.
 
-An administrator typically requires that Security Keys registered by users for the WebAuthn passwordless authentication meet different requirements. For example, the security keys may require users to authenticate to the security key using a PIN, or the security key attests with a stronger certificate authority.
+An administrator typically requires that Passkeys registered by users for the WebAuthn passwordless authentication meet different requirements. For example, the passkeys may require users to authenticate to the passkey using a PIN, or the passkey attests with a stronger certificate authority.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:194]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 15]
```diff
@@ -201,7 +201,7 @@ Set up WebAuthn passwordless support as follows:
 
 . (if not already present) Register a new required action for WebAuthn passwordless support. Use the steps described in <<_webauthn-register, Enable WebAuthn Authenticator Registration>>. Register the `Webauthn Register Passwordless` action.
 
-. Configure the policy. You can use the steps and configuration options described in <<_webauthn-policy, Managing Policy>>. Perform the configuration in the Admin Console in the tab *WebAuthn Passwordless Policy*. Typically the requirements for the security key will be stronger than for the two-factor policy. For example, you can set the *User Verification Requirement* to *Required* when you configure the passwordless policy.
+. Configure the policy. You can use the steps and configuration options described in <<_webauthn-policy, Managing Policy>>. Perform the configuration in the Admin Console in the tab *WebAuthn Passwordless Policy*. Typically the requirements for the passkey will be stronger than for the two-factor policy. For example, you can set the *User Verification Requirement* to *Required* when you configure the passwordless policy.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:204]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 16]
```diff
@@ -225,7 +225,7 @@ You can now add *WebAuthn Register Passwordless* as the required action to a use
...
 
-An administrator typically requires that Security Keys registered by users for the WebAuthn loginless authentication meet different requirements. Loginless authentication requires users to authenticate to the security key (for example by using a PIN code or a fingerprint) and that the cryptographic keys associated with the loginless credential are stored physically on the security key. Not all security keys meet that kind of requirements. Check with your security key vendor if your device supports 'user verification' and 'discoverable credential'. See <<_webauthn-supported-keys, Supported Security Keys>>.
+An administrator typically requires that Passkeys registered by users for the WebAuthn loginless authentication meet different requirements. Loginless authentication requires users to authenticate to the passkey (for example by using a PIN code or a fingerprint) and that the cryptographic keys associated with the loginless credential are stored physically on the passkey. Not all passkeys meet that kind of requirements. Check with your passkey vendor if your device supports 'user verification' and 'discoverable credential'. See <<_webauthn-supported-keys, Supported Passkeys>>.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:228]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 17]
```diff
@@ -236,7 +236,7 @@ Set up WebAuthn Loginless support as follows:
...
+. Configure the `WebAuthn Passwordless Policy`. Perform the configuration in the Admin Console, `Authentication` section,  in the tab `Policies` ->  `WebAuthn Passwordless Policy`. You have to set *User Verification Requirement* to *required* and *Require Discoverable Credential* to *Yes* when you configure the policy for loginless scenario. Note that since there isn't a dedicated Loginless policy it won't be possible to mix authentication scenarios with user verification=no/discoverable credential=no and loginless scenarios (user verification=yes/discoverable credential=yes). Storage capacity is usually very limited on passkeys meaning that you won't be able to store many discoverable credentials on your passkey.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:239]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 18]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
 
-You can now add the required action `WebAuthn Register Passwordless` to a user, already known to {project_name}, to test this. The user with the required action configured will have to authenticate (with a username/password for example) and will then be prompted to register a security key to be used for loginless authentication.
+You can now add the required action `WebAuthn Register Passwordless` to a user, already known to {project_name}, to test this. The user with the required action configured will have to authenticate (with a username/password for example) and will then be prompted to register a passkey to be used for loginless authentication.
 
 ===== Vendor specific remarks
 
 ====== Compatibility check list
 
-Loginless authentication with {project_name} requires the security key to meet the following features
+Loginless authentication with {project_name} requires the passkey to meet the following features
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:254]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 19]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
 
 ** FIDO2 compliance: not to be confused with FIDO/U2F
-** User verification: the ability for the security key to authenticate the user (prevents someone finding your security key to be able to authenticate loginless and passwordless)
-** Discoverable Credential: the ability for the security key to store the login and the cryptographic keys associated with the client application
+** User verification: the ability for the passkey to authenticate the user (prevents someone finding your passkey to be able to authenticate loginless and passwordless)
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:257]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 20]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
+** Discoverable Credential: the ability for the passkey to store the login and the cryptographic keys associated with the client application
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:258]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 21]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
 
 ====== Windows Hello
 
-To use Windows Hello based credentials to authenticate against {project_name}, configure the *Signature Algorithms* setting of the `WebAuthn Passwordless Policy` to include the *RS256* value. Note that some browsers don't allow access to platform security key (like Windows Hello) inside private windows.
+To use Windows Hello based credentials to authenticate against {project_name}, configure the *Signature Algorithms* setting of the `WebAuthn Passwordless Policy` to include the *RS256* value. Note that some browsers don't allow access to platform passkey (like Windows Hello) inside private windows.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:262]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 22]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
 
 [[_webauthn-supported-keys]]
-====== Supported security keys
+====== Supported passkeys
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:265]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 23]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
...
 
-The following security keys have been successfully tested for loginless authentication with {project_name}:
+The following passkeys have been successfully tested for loginless authentication with {project_name}:
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc:267]: I fixed it.

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnIdlessTest.java [hunk 24]
```diff
@@ -265,7 +265,7 @@ protected void checkTryAnotherWay() {
         assertThat(selectAuthenticatorPage.getLoginMethodHelpText(SelectAuthenticatorPage.USERNAME),
                 is("Start sign in by entering your username"));
         assertThat(selectAuthenticatorPage.getLoginMethodHelpText(SelectAuthenticatorPage.SECURITY_KEY),
-                is("Use your security key for passwordless sign in."));
+                is("Use your passkey for passwordless sign in."));
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnIdlessTest.java:268]: I fixe it.

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java [hunk 25]
```diff
@@ -291,7 +291,7 @@ public void webAuthnPasswordlessAlternativeWithWebAuthnAndPassword() throws IOEx
 
             selectAuthenticatorPage.assertCurrent();
             assertThat(selectAuthenticatorPage.getLoginMethodHelpText(SelectAuthenticatorPage.SECURITY_KEY),
-                    is("Use your security key for passwordless sign in."));
+                    is("Use your passkey for passwordless sign in."));
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java:294]: I fixed it.

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java (hunk 1 above)
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/WebAuthnRegisterAndLoginTest.java:265]: I fixed it.

File: testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java (hunk 2 above)
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/other/webauthn/src/test/java/org/keycloak/testsuite/webauthn/account/WebAuthnSigningInTest.java:412]: I fixed it.

File: themes/src/main/resources/theme/base/login/messages/messages_en.properties (hunk 4 above)
Comment (tnorimat) [reply themes/src/main/resources/theme/base/login/messages/messages_en.properties:479]: I fixed it.

File: themes/src/main/resources/theme/keycloak.v2/account/messages/messages_en.properties (hunk 5 above)
Comment (tnorimat) [reply themes/src/main/resources/theme/keycloak.v2/account/messages/messages_en.properties:112]: I fixed it.

File: docs/documentation/server_admin/topics/authentication/webauthn.adoc [hunk 26]
```diff
@@ -245,26 +245,26 @@ The final configuration of the flow looks similar to this:
 .LoginLess flow
 image:images/webauthn-loginless-flow.png[LoginLess flow]
 
-You can now add the required action `WebAuthn Register Passwordless` to a user, already known to {project_name}, to test this. The user with the required action configured will have to authenticate (with a username/password for example) and will then be prompted to register a security key to be used for loginless authentication.
+You can now add the required action `WebAuthn Register Passwordless` to a user, already known to {project_name}, to test this. The user with the required action configured will have to authenticate (with a username/password for example) and will then be prompted to register a passkey to be used for loginless authentication.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/webauthn.adoc]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc [hunk 27]
```diff
@@ -66,18 +66,18 @@ image:images/account-console-signing-in.png[Signing in]
...
 +
 .Signing In
-image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Security Key]
+image:images/account-console-signing-in-webauthn-2factor.png[Signing In With Passkey]
 
-. Prepare your WebAuthn Security Key. How you prepare this key depends on the type of WebAuthn security key you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
+. Prepare your WebAuthn Passkey. How you prepare this key depends on the type of WebAuthn passkey you use. For example, for a USB based Yubikey, you may need to put your key into the USB port on your laptop.
 
-. Click *Register* to register your security key.
+. Click *Register* to register your passkey.
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc]: I fixed it.

File: docs/documentation/server_admin/topics/account.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/server_admin/topics/account.adoc]: I updated the two screenshots.

**Comment by tnorimat** [comment]
@jonkoops Thank you for your review comments. I incorporated all of your review comments into the PR. Could you check them?

**Review by jonkoops — DISMISSED** [review]
LGTM!

**Review by mposolda — DISMISSED** [review]
@tnorimat @jonkoops Thanks for the updates and review! I hope to merge once tests are OK

**Comment by mposolda** [comment]
@tnorimat Could you please rebase? That should help with fixing the test failure in FIPS tests (it was fixed today in Keycloak main).

**Comment by tnorimat** [comment]
@mposolda I rebased the PR on the main branch.

**Review by ahus1 — APPROVED** [review]
re-approving after @mposolda's review and rebase

**Comment by ahus1** [comment]
@jonkoops - could you please re-approve after the rebase? Thanks!

**Review by mposolda — APPROVED** [review]
@tnorimat Thanks for the rebase!

**Pull request merged**